forms-flow-data-layer/src/middlewares/auth.py

```diff
@@ -0,0 +1,84 @@
+from typing import Any, List, Union
+
+import strawberry
+from starlette.requests import Request
+from starlette.websockets import WebSocket
+from strawberry.permission import BasePermission, PermissionExtension
+
+from graphql import GraphQLError
+from src.config.envs import ENVS
+from src.utils import KeycloakOIDC, UserContext, get_logger
+
+logger = get_logger("middlewares.auth")
+keycloak_validator = KeycloakOIDC(
+    jwks_url=ENVS.JWT_OIDC_JWKS_URI,
+    issuer=ENVS.JWT_OIDC_ISSUER,
+    audience=ENVS.JWT_OIDC_AUDIENCE,
+)
+
+
+class IsAuthenticated(BasePermission):
+    """Checking user authentication."""
+
+    message = "User is not authenticated"
+    error_extensions = {"code": "UNAUTHORIZED"}
+
+    async def has_permission(
+        self, source: Any, info: strawberry.Info, **kwargs
+    ) -> bool:
+        """Has permission is default function to check the permission."""
+        try:
+            logger.info("Fetching token from Header")
+            request: Union[Request, WebSocket] = info.context["request"]
+            auth = request.headers.get("Authorization")
+            if not auth or not auth.startswith("Bearer "):
+                return False
+            token = auth.split(" ")[1]
+            payload = await keycloak_validator.verify_token(token)
+            # Attach token info to context
+            info.context["user"] = UserContext(token=token, token_info=payload)
+            return True
+        except Exception as e:
+            logger.error(e)
+            raise GraphQLError(f"Unexpected error: {str(e)}")
+
+
+class HasAnyRole(BasePermission):
+    """Class for check authorization."""
+
+    message = "User role not exist"
+    error_extensions = {"code": "UNAUTHORIZED"}
+
+    def __init__(self, roles: List):
+        self.roles = roles
+
+    async def has_permission(
+        self, source: Any, info: strawberry.Info, **kwargs
+    ) -> bool:
+        try:
+            user_rols = info.context["user"].roles
```

Comment:
Can we check the roles as in the user_context file in utils ?

Reply:
resolved